### PR TITLE
Fix ReSpec definitions of RemotePlayback interface and other issues.

### PR DESCRIPTION
**Pull request by markafoltz**
Converts references to terms to Bikeshed syntax.

It also:
- Converts the use of `void` to `undefined` which was flagged by ReSpec
- Adds a `group` tag to the ReSpec metadata
- Stops linking `user agent` because we all know what that means

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/remote-playback/pull/151.html" title="Last updated on Sep 29, 2022, 8:50 PM UTC (c6601d0)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/remote-playback/151/74cd762...c6601d0.html" title="Last updated on Sep 29, 2022, 8:50 PM UTC (c6601d0)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -146,7 +146,7 @@
     <section id='conformance'>
       <p>
         This specification defines conformance criteria that apply to a single
-        product: the <dfn>user agent</dfn> that implements the interfaces that
+        product: the user agent that implements the interfaces that
```
Comment (tidoust) [inline index.html:149]: What's the motivation behind dropping the definition? If it's the fear of creating a definition that gets exported outside of the Remote Playback API, I note that would not happen: IDL names are exported by default, but `<dfn>` are only exported when they have a `class="export"` attribute.

File: index.html [hunk 2]
```diff
@@ -188,7 +188,7 @@ <h2>
         playback device</a>. In such a case, both the <a>browsing context</a>
         and the media player run on the same UA and the operating system is
         used to route the player output to the <a>remote playback device</a>.
-        This is commonly referred to as the <dfn><b id="media-mirroring">media
+        This is commonly referred to as the <dfn><b>media
         mirroring</b></dfn> case. This specification imposes no requirements on
```
Comment (tidoust) [inline index.html]: Definitions already appear as bold terms. While we're at it, I'm wondering whether we can also drop the extra bold:

```suggestion
        This is commonly referred to as the <dfn>media
        mirroring</dfn> case. This specification imposes no requirements on
```

File: index.html [hunk 3]
```diff
@@ -197,20 +197,20 @@ <h2>
         communicate with the <a>browsing context</a> but is unable to fetch the
         media, the <a>browsing context</a> needs to fetch the media data and
         pass it on to the <a>remote playback device</a> for rendering. This is
-        commonly referred to as <dfn><b id="media-remoting">media
+        commonly referred to as <dfn><b>media
         remoting</b></dfn> case.
```
Comment (tidoust) [inline index.html]:
```suggestion
        commonly referred to as <dfn>media remoting</dfn> case.
```

File: index.html [hunk 4]
```diff
@@ -197,20 +197,20 @@ <h2>
...
         context</a> does not need to fetch or render the remoted media. In this
         case, the UA acts as a proxy that requests the <a>remote playback
         device</a> to play the media itself by passing the necessary data like
-        the media source. This is commonly referred to as the <dfn><b id=
-        "media-flinging">media flinging</b></dfn> case. This way of attaching
-        to displays could be enhanced in the future by defining a standard
-        protocol for delivering these types of messages that remote playback
-        devices could choose to implement.
+        the media source. This is commonly referred to as the <dfn><b>media
+        flinging</b></dfn> case. This way of attaching to displays could be
```
Comment (tidoust) [inline index.html]:
```suggestion
        the media source. This is commonly referred to as the <dfn>media
        flinging</dfn> case. This way of attaching to displays could be
```

File: index.html [hunk 5]
```diff
@@ -390,7 +390,7 @@ <h3>
       </section>
       <section>
         <h3>
-          <a>`RemotePlayback`</a> interface
+          {{RemotePlayback}} interface
```
Comment (tidoust) [inline index.html]: If we get back to definitions in the prose, I would actually propose to have the definition target the heading here:

```suggestion
          <dfn>RemotePlayback</dfn> interface
```

**Review by tidoust — COMMENTED** [review]
No problem with most of the simplifications to use more modern ReSpec shorthands in that pull request.

That said, I see the PR also updates the spec so that the IDL definitions end up in the IDL block. That is not the recommended way. At least, that's not the [recommended approach in the ReSpec Documentation](https://respec.org/docs/#ideal-linking-setup) (FWIW, the Presentation API spec follows that recommendation).

You mention #137 but that change is not going to have any impact from a cross-reference perspective. Whether definitions are in an IDL block or in the prose, ReSpec understand they are IDL terms and will export them with the right attributes so that they can be picked up by tools that creates cross-references databases. Typically, IDL terms are already in [ReSpec's cross-ref database](https://respec.org/xref/?term=RemotePlayback) and you can already use `{{RemotePlayback}}` from another ReSpec spec. These terms cannot be used from within a Bikeshed spec for now because the Remote Playback API is not part of Shepherd's crawl. Nothing prevents that and I requested addition some time ago in https://github.com/tabatkins/bikeshed/issues/1664

In other words, I would roll back on the updates that convert `<dfn> </dfn>` to `{{ }}` although you can indeed simplify them a bit (dropping useless backticks notably, e.g. `<dfn data-dfn-for="RemotePlayback">watchAvailability()</dfn>` should be enough).

Or are the changes intended because you prefer things that way?

File: index.html (hunk 5 above)
Comment (markafoltz) [reply index.html]: I'm can do that but to be honest am a little confused about how ReSpec treats definitions.   ReSpec already generates a `<dfn>` tag for RemotePlayback from the WebIDL:

```
<dfn data-export="" data-dfn-type="interface" id="dom-remoteplayback" data-idl="interface" data-title="RemotePlayback" data-dfn-for="" class="idlID" tabindex="0" aria-haspopup="dialog"><code>RemotePlayback</code></dfn>
```

How would ReSpec (or the definition exporter for cross-referencing) know which definition to use if there are multiple?

File: index.html (hunk 1 above)
Comment (markafoltz) [reply index.html:149]: I looked at a sample of other specs and none of them define or reference user agent defined elsewhere.

Moreover, this spec is not attempting to define what a "user agent" is.

**Comment by markafoltz** [comment]
> No problem with most of the simplifications to use more modern ReSpec shorthands in that pull request.
> 
> That said, I see the PR also updates the spec so that the IDL definitions end up in the IDL block. That is not the recommended way. At least, that's not the [recommended approach in the ReSpec Documentation](https://respec.org/docs/#ideal-linking-setup) (FWIW, the Presentation API spec follows that recommendation).

I can try to follow that setup but I would like to understand how it works first; the ReSpec guide doesn't seem to explain how you can have multiple definitions of the same term.
 
> You mention #137 but that change is not going to have any impact from a cross-reference perspective. Whether definitions are in an IDL block or in the prose, ReSpec understand they are IDL terms and will export them with the right attributes so that they can be picked up by tools that creates cross-references databases. Typically, IDL terms are already in [ReSpec's cross-ref database](https://respec.org/xref/?term=RemotePlayback) and you can already use `{{RemotePlayback}}` from another ReSpec spec. These terms cannot be used from within a Bikeshed spec for now because the Remote Playback API is not part of Shepherd's crawl. Nothing prevents that and I requested addition some time ago in [tabatkins/bikeshed#1664](https://github.com/tabatkins/bikeshed/issues/1664)

From the issue it sounded like we shouldn't submit this spec for cross referencing yet because of issues with term definitions like `RemotePlayback`; @tabatkins says you can't define it with a plain `<dfn>`, which is what this PR is trying to resolve.

File: index.html (hunk 5 above)
Comment (tidoust) [reply index.html]: There would not be multiple definitions. ReSpec only generates a `<dfn>` tag from the WebIDL if it does not already exist in the prose. It will rather create `<a>` throughout the WebIDL block when the prose contains the relevant definitions.

**Comment by tidoust** [comment]
> I can try to follow that setup but I would like to understand how it works first; the ReSpec guide doesn't seem to explain how you can have multiple definitions of the same term.

You cannot but there won't be multiple definitions of the same term. ReSpec is smart enough to generate `<a>` and not `<dfn>` in WebIDL blocks when the definitions already appear elsewhere in the spec.

> From the issue it sounded like we shouldn't submit this spec for cross referencing yet because of issues with term definitions like `RemotePlayback`; @tabatkins says you can't define it with a plain `<dfn>`, which is what this PR is trying to resolve.

That's no longer relevant. Shepherd could not crawl the /TR version because it was too old at the time and did not yet follow the definitions data model. Also, since Shepherd does not run JavaScript, it cannot crawl raw (as opposed to generated versions of) ReSpec specs, so our Editor's Draft could not be added either. Since then, the /TR version was updated (and now matches the Editor's Draft) and we've been publishing the generated version of the spec to GitHub Pages in any case. Shepherd can crawl either version without problems.

FYI, there are a minority of specs that don't follow the "definitions in prose" pattern and rather have the definitions in the WebIDL block, or that use a mix of both approaches (for instance, the HTML spec has the interface definitions in the WebIDL blocks but methods definitions in the prose). That's why I suggested that this was up to you ;)

Note I'm happy to volunteer and update the PR to what I think would be the right result if you'd like me to!

**Comment by markafoltz** [comment]
OK, converted to define WebIDL terms in prose.  Is there anything we can do to expedite the term indexing in Shepherd?

File: index.html (hunk 5 above)
Comment (markafoltz) [reply index.html]: OK, please check the current PR which follows the suggested pattern.

**Review by tidoust — APPROVED** [review]
> Is there anything we can do to expedite the term indexing in Shepherd?

In theory, that should be a matter of adding a line somewhere in the right Shepherd table. I pinged Tab and Peter on the issue.
